### ISSUE TITLE
Marathon demo doesn't work

### ISSUE THREAD

**Issue by gengmao** [alde/eremetic#36]
I deployed "alde/eremetic:0.8.0-9-g2a09a33" via Marathon (following the https://github.com/alde/eremetic/blob/master/misc/eremetic.json). 

I can see it runs in Marathon, however, Mesos keeps terminating the framework it registried. The task I posted to Eremetic never got executed. The task is stuck in "Staging" status. 

Here is the mesos-master log. 
```
I1208 01:01:08.805589  3046 master.cpp:2179] Received SUBSCRIBE call for framework 'Eremetic' at scheduler(1)@172.17.0.3:53366
I1208 01:01:08.805904  3046 master.cpp:2250] Subscribing framework Eremetic with checkpointing disabled and capabilities [  ]
I1208 01:01:08.807024  3047 master.cpp:4967] Sending 7 offers to framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
I1208 01:01:11.802533  3048 master.cpp:1119] Framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366 disconnected
I1208 01:01:11.802851  3048 master.cpp:2475] Disconnecting framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
I1208 01:01:11.802917  3048 master.cpp:2499] Deactivating framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803131  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803131  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803346  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803346  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803606  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803606  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803771  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803771  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803921  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.803921  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.804134  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.804134  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.804277  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
W1208 01:01:11.804277  3048 master.hpp:1532] Master attempted to send message to disconnected framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
I1208 01:01:11.804376  3048 master.cpp:1143] Giving framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366 0ns to failover
I1208 01:01:11.806012  3050 master.cpp:4815] Framework failover timeout, removing framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
I1208 01:01:11.806071  3050 master.cpp:5571] Removing framework 68ee186a-6e6d-45ce-a411-f9baafac04f8-0038 (Eremetic) at scheduler(1)@172.17.0.3:53366
```
Here are the error logs of Eremetic
```
I1208 00:45:24.127424 13380 exec.cpp:134] Version: 0.25.0
I1208 00:45:24.129081 13385 exec.cpp:208] Executor registered on slave 6e9a965d-2507-4e91-8d80-55a9d5f2d9e4-S1
WARNING: Your kernel does not support swap limit capabilities, memory limited without swap.
ERROR: logging before flag.Parse: W1208 00:45:24.600565       1 scheduler.go:177] Failed to obtain username: user: Current not implemented on linux/amd64
ERROR: logging before flag.Parse: I1208 00:45:24.601538       1 scheduler.go:323] Initializing mesos scheduler driver
ERROR: logging before flag.Parse: I1208 00:45:24.601694       1 scheduler.go:792] Starting the scheduler driver...
ERROR: logging before flag.Parse: I1208 00:45:24.601793       1 http_transporter.go:407] listening on 0.0.0.0 port 53366
ERROR: logging before flag.Parse: I1208 00:45:24.601906       1 scheduler.go:809] Mesos scheduler driver started with PID=scheduler(1)@172.17.0.3:53366
ERROR: logging before flag.Parse: I1208 00:45:24.601974       1 scheduler.go:999] Scheduler driver running.  Waiting to be stopped.
2015/12/08 00:45:24 Connected to 172.16.42.171:2181
2015/12/08 00:45:24 Authenticated: id=5283109436024422411, timeout=40000
ERROR: logging before flag.Parse: I1208 00:45:24.613195       1 scheduler.go:374] New master master@172.16.32.55:5050 detected
ERROR: logging before flag.Parse: I1208 00:45:24.613223       1 scheduler.go:435] No credentials were provided. Attempting to register scheduler without authentication.
```

I found mesos master couldn't ping to 172.17.0.3, which is the ip Eremetic saw inside the container. 

Also I found the /marathon.sh seems didn't resolve the MESSENGER_ADDRESS AND MESSENGER_PORT correctly... They are  
```
MESSENGER_ADDRESS=172.16.5.91 ip-172-16-5-91.us-west-2.compute.internal
MESSENGER_PORT=
```
Here are the environment variables in the Eremetic container:
```
HOSTNAME=3147495689df
SHLVL=1
OLDPWD=/root
HOME=/root
PORT=8000
MESOS_CONTAINER_NAME=mesos-6e9a965d-2507-4e91-8d80-55a9d5f2d9e4-S1.9ed474b4-cc2f-4207-b887-8a07252109a3
MARATHON_APP_ID=/eremetic
PORTS=31683
PORT0=31683
MASTER=zk://172.16.42.171:2181,172.16.18.113:2181,172.16.6.225:2181/mesos
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
MESOS_SANDBOX=/mnt/mesos/sandbox
HOST=ec2-xx-xx-xx-xx.us-west-2.compute.amazonaws.com
PORT_8000=31683
MARATHON_APP_VERSION=2015-12-08T00:45:16.866Z
PWD=/
ADDRESS=0.0.0.0
MESOS_TASK_ID=eremetic.f44e57d4-9d44-11e5-b201-024290fce437
```

Don't know how 53366 port came from... Do you have any clues? 

I will tweak a bit marathon.sh and eremetic.json then see. Thanks.

**Comment by keis** [comment]
Yes, this can happen if eremetic registering with the local ip of the docker container as you discovered. Which typically means $HOST can't be resolved from within the container. Or maybe in this case that the output from nslookup is not parsed correctly.
Is the 172.X.X.X address a public one you could be using from the mesos master?

The marathon config should have entries for two randomly selected ports but it seems you're only getting the one for HTTP. The 53366 is randomly selected by eremetic when no port is specified.

**Comment by gengmao** [comment]
Thanks @keis. 

My bad about missing `MESSENGER_PORT` - I didn't understand `{ "containerPort": 0, "hostPort": 0 }` so removed it from the json file. After adding it back I can see `MESSENGER_PORT` was set correctly. 

The problem about `MESSENGER_ADDRESS` is the nslookup output inside eremetic container. It is like below: 
```
/opt/eremetic # nslookup $HOST
Server:    (null)
Address 1: ::1 localhost
Address 2: 127.0.0.1 localhost

Name:      ec2-xxx-xxx-xxx-xxx.us-west-2.compute.amazonaws.com (masked here)
Address 1: 172.16.5.11 ip-172-16-5-11.us-west-2.compute.internal
```

`172.16.5.11` is a private ip which mesos master can access. I don't understand why there is another address called `ip-172-16-5-11.us-west-2.compute.internal` (maybe it is related to alpine:3.2? I didn't see it in ubuntu host's nslookup outoput), but I figured out a small change to marathon.sh to extract only the first part. I will submit a PR soon.

**Comment by keis** [comment]
I assume this is fixed now that #38 is merged. Please re-open if not.

**Comment by gengmao** [comment]
Yes its fixed. Thanks for help!
